src/tbp/monty/frameworks/config_utils/make_dataset_configs.py

```diff
@@ -1102,3 +1104,279 @@ class FiveLMMountConfig:
+
+    return positions
+
+
+def make_multi_sensor_mount_config(
+    n_sensors: int,
+    agent_id: str = "agent_id_0",
+    sensor_ids: Optional[Sequence[str]] = None,
+    height: Number = 0.0,
+    position: ArrayLike = (0, 1.5, 0.2),  # agent position
+    resolutions: Optional[ArrayLike] = None,
+    positions: Optional[ArrayLike] = None,
+    rotations: Optional[ArrayLike] = None,
+    semantics: Optional[ArrayLike] = None,
+    zooms: Optional[ArrayLike] = None,
+) -> Mapping[str, Any]:
+    """Generate a multi-sensor mount configuration.
+
+    Creates a multi-sensor, single-agent mount config. Its primary use is in generating
+    a `MultiLMMountHabitatDatasetArgs` config. This function assumes that a view finder
+    will be included which entails that the number of sensor_ids, resolution, etc. will
+    be of length `n_sensors + 1`. Defaults are reasonable and reflect current common
+    practices.
+
+    Args:
+        n_sensors: Number of sensors. Count should not include an optional view finder.
+        agent_id: ID of the agent. Defaults to "agent_id_0".
+        sensor_ids: IDs of the sensor modules. Defaults to
+            `["patch_0", "patch_1", ... "patch_{n_sms - 1}", "view_finder"]`.
+        height: Height of the agent. Defaults to 0.0.
+        position: Position of the agent. Defaults to [0, 1.5, 0.2].
+        resolutions: Resolutions of the sensors. Defaults to (64, 64) for all
+            sensors.
+        positions: Positions of the sensors. If not provided, calls
+            `make_sensor_positions_on_grid` with its default arguments.
+        rotations: Rotations of the sensors. Defaults to [1, 0, 0, 0] for all sensors.
+        semantics: Defaults to `True` for all sensors.
+        zooms: Zooms of the sensors. Defaults to 10.0 for all sensors except for the
+          except for the view finder (which has a zoom of 1.0)
+
+    Returns:
+        dict: A dictionary representing a complete multi-sensor mount config.
+    """
+    assert n_sensors > 0, "n_sensors must be a positive integer"
+    arr_len = n_sensors + 1
+
+    # Initialize with agent info, then add sensor info.
+    mount_config = {
+        "agent_id": str(agent_id),
+        "height": float(height),
+        "position": np.asarray(position),
+    }
+
+    # sensor IDs.
+    if sensor_ids is None:
+        sensor_ids = np.array(
+            [f"patch_{i}" for i in range(n_sensors)] + ["view_finder"], dtype=object
+        )
+    else:
+        sensor_ids = np.array(sensor_ids, dtype=object)
+    assert sensor_ids.shape == (arr_len,), "`sensor_ids` must have length n_sensors + 1"
+    mount_config["sensor_ids"] = sensor_ids
+
+    # sensor resolutions
+    if resolutions is None:
+        resolutions = np.full([n_sensors + 1, 2], 64)
+    else:
+        resolutions = np.asarray(resolutions)
+    assert resolutions.shape == (
+        arr_len,
+        2,
+    ), "`resolutions` must have shape (n_sensors + 1, 2)"
```

Comment:
Depending on how you address the default view-finder thing, you can change the message in these asserts to use an "f-string" formatting that reflects the parameters.

Reply:
Got it.